### ISSUE TITLE
Probably false detection of "MD007 Unordered list indentation"

### ISSUE THREAD

**Issue by N0rbert** [markdownlint/markdownlint#313]
### Replication Case

Let's assume that we have code presented below:

```
# First header

Some text before list:

* orange;
* yellow.

Some steps:

1. List of colors:

    * Red
    * Green
    * Yellow

1. Third line
```

it renders normally here on GitHub and at least in ReText, MarkText, GhostWriter and RStudio.

### Environment

Ubuntu 16.04 LTS was used with

**MDL Version**
0.9.0 as Snap

### Expected Behavior

`mdl` does not show violation of any rules.

### Actual Behavior

`mdl` shows the following:

> lists.md:12: MD007 Unordered list indentation

Line 12 is `* Red` here. 
It is nested inside numbered list item. 
It does not have any connection with `* orange` or `* yellow`.
So I think it is a false detection.

**Comment by yawboateng** [comment]
am having the same issue. Even though the error is on the nested list of line 12 (`* Red`), i notice if i change the unordered list `* orange` and `* yellow` to  ordered list `1. orange` and `2. yellow`, it works fine and passes

Fails on the indented list on `* Red`, with error message `MD007 Unordered list indentation`: 
```
# First header

Some text before list:

* orange;
* yellow.

Some steps:

1. List of colors:

    * Red
    * Green
    * Yellow

2. Third line
```

Works fine:
```
# First header

Some text before list:

1. orange;
2. yellow.

Some steps:

1. List of colors:

    * Red
    * Green
    * Yellow

2. Third line
```

**Comment by N0rbert** [comment]
Above is a silly minimal reproducible example, the real content of my document was different. I do not plan to change writing style to satisfy the linter, even if linter is good overall :)

**Comment by yawboateng** [comment]
> Above is a silly minimal reproducible example, the real content of my document was different. I do not plan to change writing style to satisfy the linter, even if linter is good overall :)


i know! My content is different as well, however similar structure as the example you posted and same error.
I'm not asking you to change your writing style to satisfy the linter  :expressionless:.... i'm just reiterating that I also hit the bug and my work around for it - might help the devs debug and fix it :)

**Comment by ghost** [comment]
Any updates on this? I'm also experiencing this bug

**Comment by jaymzh** [comment]
Nope, sorry, I have been slammed at work and haven't had time to work on this.

**Comment by ghost** [comment]
No problem, thanks for the update!

**Comment by N0rbert** [comment]
Still happens with document from the first comment.

**Comment by N0rbert** [comment]
```
cat <<EOF > test.md
# First header

Some text before list:

* orange;
* yellow.

Some steps:

1. List of colors:

    * Red
    * Green
    * Yellow

1. Third line
EOF
```


mdl 0.12 is still affected by this bug:

```
$ mdl test.md 
test.md:12: MD007 Unordered list indentation

A detailed description of the rules is available at https://github.com/markdownlint/markdownlint/blob/master/docs/RULES.md
$ mdl --version
0.12.0
```

While remark shows no problems with above file:

```
$ remark --no-stdout test.md 
test.md: no issues found

$ remark --version
remark: 14.0.2, remark-cli: 11.0.0
```

Please fix this issue in the master branch of `mdl`.

**Comment by jaymzh** [comment]
The issue is still open. What made you believe it is fixed?

I haven't had much time to work on this project but I have been reviewing PRs as they come in. If you want to submit one I'll be happy to review it.

**Comment by mjbear** [comment]
> The issue is still open. What made you believe it is fixed?

@jaymzh
Nice to meet you.
I believe @N0rbert was saying that a different Markdown linter ([remark](https://github.com/remarkjs/remark-lint)) lints fine.

I've been fiddling with other Markdown linters and [DA's Node.js md linter](https://github.com/DavidAnson/markdownlint) parses ok.
 
> I haven't had much time to work on this project but I have been reviewing PRs as they come in. If you want to submit one I'll be happy to review it.

I've hit this issue as well.
I'm not a Rubyist but I'll see if I can figure out how to go about fixing this.
